GDJS/Runtime/variable.ts

```diff
@@ -211,73 +284,197 @@ namespace gdjs {
      * @param newValue The new string to be set
      */
     setString(newValue: string): void {
-      this._str = newValue;
-      this._numberDirty = true;
-      this._stringDirty = false;
+      this._type = 'string';
+      this._str = '' + newValue;
+    }
+
+    /**
+     * Get the value of the variable, considered as a boolean
+     * @return The boolean value of the variable.
+     */
+    getAsBoolean(): boolean {
+      if (this._type !== 'boolean') {
+        if (this._type === 'number') return this._value !== 0;
+        else if (this._type === 'string')
+          return this._str !== '0' && this._str !== '' && this._str !== 'false';
```

Comment:
Do you think we should just do `this._str !== ''`, like in JavaScript? 

I know it's "convenient" to cast `"false"` to `false`, but I wonder if it's not more surprising than anything else. No other language is really doing this?